src/main/java/sirius/biz/process/logs/ProcessLog.java

```diff
@@ -393,6 +395,31 @@ public ProcessLog withFormattedMessage(String message, Object... parameters) {
         return withMessage(Strings.apply(message, parameters));
     }
 
+    /**
+     * Specifies an exception from where to inherit the message.
+     * <p>
+     * If the provided exception is a {@link HandledException}, also considers the filled
+     * {@link ProcessContext#HINT_MESSAGE_TYPE} to categorize the log in message types and
+     * {@link ProcessContext#HINT_MESSAGE_COUNT} to limit the amount of allowed messages per type.
+     *
+     * @param exception the {@link Exception} to retrieve the message and eventually hints from
+     * @return the log entry itself for fluent method calls
+     */
+    public ProcessLog withException(Exception exception) {
```

Comment:
Either limit this to HandledException or move the Exceptions.handle(...) from ProcessEnvironment to here...